syntax = "proto3";  // 指定语法格式，注意 proto3 不再支持 proto2 的 required 和 optional

import "proto/user.proto";
package notebook;   // 指定生成的 user.pb.go 的包名，防止命名冲突


// service 定义开放调用的服务，即 NotebookService 微服务
service Notebook {
    // rpc 定义服务内的 GetNotebookInfo 远程调用
    //  方法名  方法参数                 返回值
    /*rpc GetNotebookInfo (NotebookRequest) returns (NotebookResponse);*/
    rpc GetNotebookInfo (GetNoteBookRequest) returns (GetNoteBookResponse);
}


message GetNoteBookRequest {
    string user_id = 1;
}

message MsgNoteBook {
    string title = 1;     // 标题
    string summary = 2;   // 概述
    string image = 3;     // 图片
    string userId = 4;   // 用户 id
    string id = 5;        // notebook id
}

message GetNoteBookResponse {
    // repeated 修饰符表示字段是可变数组，即 slice 类型
    repeated MsgNoteBook data = 1;
    user.MsgUser user = 2;
}

message ErrorResponse {
    string code = 1;        // 错误码
    string message = 2;     // message
    string data = 3;        // data
}